obj.js

```diff
@@ -300,20 +349,24 @@ var gLocalClasses = new Map();
 async function createObjectDescription(obj, id) {
   let className = obj.constructor.name;
   assert(className, "Could not find class name for local object");
-  if ( !gLocalClasses.get(className)) {
+  if ( !gLocalClasses.has(className)) {
     await sendClassDescription(className, obj);
   }
 
   let props = null;
-  for (let propName in obj) {
-    if (propName.startsWith("_") ||
-        typeof(obj[propName]) == "function") {
+  for (let propName of Object.getOwnPropertyNames(obj)) {
```

Comment:
What about class hierarchies with inheritance? I think this would ignore all properties of parent classes.

Reply:
I obviously misunderstood what this was trying to do - for some reason I thought this was trying to serialise the object, not its class, which I thought happened in `sendClassDescription`. Is the idea that you have a class prototype with a constant set on it, so that you can read its value from an instance of the class?

Reply:
It does serialize the object.

```
class Vehicle {
  wheels = 4;
}
class Car {
  seats = 4;
}

let ferrari = new Car();
ferrari.seats = 2;
```

Will `wheels` be sent?

Reply:
You're setting properties on the object, not on the class, so yes, they will be sent, as a one-time operation the first time the object is serialised.